### ISSUE TITLE
Color contrast issues when material design modal is opened

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2329]
Expectation: 
axe does not report violations for content that is covered by a modal overlay

Actual:
When I run axe on the following page, after I open a modal, contrast issues are reported:
https://material.angular.io/components/dialog/examples

**Comment by stevekeator** [comment]
I'm going to add to this since I think this is related to OP's issue, and was going to submit a report on my own this morning.

Color contrast is not scanning correctly in hidden content.  It produces many false positives and incomplete runs.

![image](https://user-images.githubusercontent.com/5422712/85863660-b8a8f980-b780-11ea-88df-5761be451450.png)

When this closed dropdown, several violations for color contrast are reported for the dropdown content.  This reports in Coconut w/ 3.5.2 (yes, I need to update) as:

![image](https://user-images.githubusercontent.com/5422712/85863828-ee4de280-b780-11ea-9259-2bb10362f42e.png)

It also reports the same in Core 3.5.5 as used in an in-page scanning module that I'm currently building for a Drupal CMS.

In addition, in many cases, with these dropdowns closed, I was seeing instances of incomplete scans for color contrast in portions of the hidden element.

It would appear that color contrast scans almost have a visual approach method to them.  I honestly don't know anything about the inner workings of Axe, but have used the library extensively for a while now on a few different projects supporting accessibility.  However this works, it doesn't seem all that of a great methodology to me.  Foreground and background colors should be detectable on the element, regardless of visibility.  If nothing is found, it should be able to start looking up the DOM hierarchy until it has values for each, and make a good guess from there.  None of the errors that I was seeing would have ever been registered if that was the case.

In my case, I'm just able to throw a line of JS before my axe.run() to open the dropdowns on the page, and that makes my problem go away.  Nevertheless, this seems like a pretty big issue.  Apologies if this has already been reported.

**Comment by stevekeator** [comment]
To be clear, the dropdowns function with CSS display properties (block/none) and NOT CSS visibility.

**Comment by straker** [comment]
I looked into this. The problem is our code purposefully filters out the modal because of `reduceToElementsBelowFloating`. The modal uses `position: fixed` so our code removes all of the modal to get to the elements below it. Since this was a way to remove fixed headers from the color-contrast check, I'm not sure how to fix it.

**Comment by straker** [comment]
Talked with Wilco, we're going to close this issue as a user testing the entire page will either see incomplete for color-contrast (the modal is on top) or failures underneath (the modal is `positioned: fixed` so we treat it as not there). If the user really wants to test just the modal they can use the modal as the context instead of the full page.

@stevekeator For your issue, would you mind opening up a separate ticket and maybe a link to the page this is happening on? Thanks.
